### PR TITLE
TG-5586 Add unit tests for com.github.theholywaffle.teamspeak3.commands.FileCommands

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
Hi,
I've analysed your codebase and seen some gaps in the coverage of `commands.FileCommands`.
I've written tests for these functions with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/datasheet).
Hopefully, these tests should help you detect regressions caused by future code changes.

### CODE REVIEW

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
+package com.github.theholywaffle.teamspeak3.commands;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class FileCommandsTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
```
Comment (petersear) [inline src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: Break line after @Rule, like the @Test entries (I have had a Maven run fail on a line like this)

**Review by sudo-jaa — APPROVED** [review]
Flawless :+1: 
Which tests did you write yourself? I can't even tell. :D

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    FileCommands.ftRenameFile("dir1", null, 001, null, 002, null);
+  }
+
+  @Test
+  public void ftRenameFileAndChannel() {
+    final String expected = "ftrenamefile cid=1 cpw= tcid=2 tcpw= oldname=\\/dir1 newname=\\/dir2";
+    Assert.assertEquals(expected, FileCommands.ftRenameFile("/dir1", "dir2", 001, null, 002, null).toString());
+    Assert.assertEquals(expected, FileCommands.ftRenameFile("dir1", "/dir2", 001, null, 002, null).toString());
+  }
+}
```
Comment (petersear) [inline src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: No newline at end of file.

**Review by paulkennethkent — COMMENTED** [review]
New line

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java (hunk 2 above)
Comment (paulkennethkent) [inline src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: New line

**Review by petersear — CHANGES_REQUESTED** [review]

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  @Test
+  public void ftCreateDir_NullPathException() {
+    thrown.expect(IllegalArgumentException.class);
+    FileCommands.ftCreateDir(null, 001, null);
+  }
+
+  @Test
+  public void ftCreateDir() {
```
Comment (paulkennethkent) [inline src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java:20]: Have you seen this style of test case before or is it something that they are using already? I feel like it doesn't follow the junit framework of ` Arrange` `Act` and `Assert Result`.

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java (hunk 1 above)
Comment (freddytuxworth) [reply src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: If it's useful, VS code will automatically add these newlines (and any other good formatting changes, probably according to the google java style guide) with Ctrl+Shift+I.

Comment (JohnLBergqvist) [reply src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: Really? @petersear Platform has never done this, and no-one's ever commented on it before...

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java (hunk 3 above)
Comment (JohnLBergqvist) [reply src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java:20]: No, I just didn't want to unnecessarily break the tests up so much. I've had comments in the past saying there's no need to have tests so long-winded.

**Review by petersear — APPROVED** [review]

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java (hunk 1 above)
Comment (petersear) [reply src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: Yes, only seen it the once. I can't remember the project now but might be able to track it down.

**Review by peterschrammel — APPROVED** [review]
Looks neat. What's the coverage on this class?

**Comment by JohnLBergqvist** [comment]
@peterschrammel Everything apart from the private method (1 line). I've heavily re-wrote these (flattening the declarations where possible), and the stuff I added from scratch was basically every test that doesn't throw an exception. And I've had to use the `.toString` method to check that the object returned by the function was created (as that class has no `.equals` or other getters I could use).

File: src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java (hunk 1 above)
Comment (petersear) [reply src/test/java/com/github/theholywaffle/teamspeak3/commands/FileCommandsTest.java]: It was the feign PR:
https://github.com/OpenFeign/feign/pull/844
mvn compile + test were fine but there was a Travis failure which was resolved by breaking the line.

**Review by paulkennethkent — APPROVED** [review]

**Pull request closed without merging**